src/pages/index.tsx

```diff
@@ -46,6 +50,13 @@ const linkTextStyles =
   'text-blue-600 dark:text-blue-300 transition hover:text-purple-600 dark:hover:text-purple-300';
 
 export default function IndexPage(): JSX.Element {
+  const firebaseUser = useFirebaseUser();
+  const loading = useIsUserDataLoaded();
+  React.useEffect(() => {
+    if (firebaseUser) {
```

Comment:
As mentioned, add some check for if the last page (page that the user navigated from) is NOT /dashboard, and then if so, we navigate.